src/dra/42398.c

```diff
@@ -2801,7 +2801,28 @@ INCLUDE_ASM("asm/dra/nonmatchings/42398", func_800FAEC4);
 // https://decomp.me/scratch/HEhaF by @pixel-stuck
 INCLUDE_ASM("asm/dra/nonmatchings/42398", func_800FAF44);
 
-INCLUDE_ASM("asm/dra/nonmatchings/42398", func_800FB004);
+void func_800FB004(void) {
+    s32 temp_a1;
+    s32 temp_v0;
+    temp_a1 = func_800FD6C4(D_801375CC);
+    if (((-D_80137688) / 12) != 0) {
+        if (*D_80137844 == 0) {
+            *D_80137844 = 1;
+        }
+    } else {
+        *D_80137844 = 0;
+    }
+
+    temp_v0 = -D_80137688 + D_8013767C;
```

Comment:
Did you try flipping the expression as `D_8013767C - D_80137688`?

Reply:
Flipping it seems to cause the loads to happen in a different order:

```
    temp_v0 = D_8013767C - D_80137688;

18ce4:    lui     a0,0x2aaa                              18ce4:    lui     a0,0x2aaa                
18ce8: ~> lui     v1,%hi(D_80137688)              r      18ce8: ~> lui     v0,%hi(D_8013767C)       
18cec:    lh      v1,%lo(D_80137688)(v1)          r      18cec:    lh      v0,%lo(D_8013767C)(v0)   
18cf0:    lui     v0,%hi(D_8013767C)              r      18cf0:    lui     v1,%hi(D_80137688)       
18cf4:    lh      v0,%lo(D_8013767C)(v0)          r      18cf4:    lh      v1,%lo(D_80137688)(v1)   
18cf8:    ori     a0,a0,0xaaab                           18cf8:    ori     a0,a0,0xaaab             
```